router/core/header_rule_engine_test.go

```diff
@@ -241,3 +323,144 @@ func TestSubgraphNamedHeaderRule(t *testing.T) {
 	assert.Empty(t, updatedClientReq2.Header.Get("X-Test-Subgraph-1"))
 	assert.Equal(t, "Test-Value-2", updatedClientReq2.Header.Get("X-Test-Subgraph-2"))
 }
+
+func TestSubgraphRenameNamedHeaderRule(t *testing.T) {
```

Comment:
Why does the branch contains changes from your work on the rename header rule?

Reply:
Actually I worked on same code base and then published the changes to the new branch and created the draft pr.